api/v1beta3/vcdcluster_webhook.go

```diff
@@ -59,15 +61,34 @@ var _ webhook.Validator = &VCDCluster{}
 func (r *VCDCluster) ValidateCreate() (admission.Warnings, error) {
 	vcdclusterlog.Info("validate create", "name", r.Name)
 
-	// TODO(user): fill in your validation logic upon object creation.
+	var errList field.ErrorList
+
+	if len(r.Spec.MultiZoneSpec.Zones) != 0 && r.Spec.Ovdc != "" {
+		errList = append(errList, field.Invalid(field.NewPath("spec", "ovdc"), r.Spec.Ovdc,
+			"field must not be set in a multi-AZ cluster"))
+	}
+
+	if len(r.Spec.MultiZoneSpec.Zones) != 0 && r.Spec.OvdcNetwork != "" {
+		errList = append(errList, field.Invalid(field.NewPath("spec", "ovdcNetwork"), r.Spec.Ovdc,
+			"field must not be set in a multi-AZ cluster"))
+	}
+
+	if len(r.Spec.MultiZoneSpec.Zones) != 0 && r.Spec.MultiZoneSpec.ZoneTopology != DCGroup {
```

Comment:
So basically this implies that the defaulting hook is called before the validating hook, correct? It seems to be the logical thing to do.

Reply:
Should we verify that if there is a zoneTopplogy (== DCGroup) then len(zones) should be > 0 ?

Reply:
> So basically this implies that the defaulting hook is called before the validating hook, correct? It seems to be the logical thing to do.

that is my understanding per the documentation

Reply:
> Should we verify that if there is a zoneTopplogy (== DCGroup) then len(zones) should be > 0 ?

I've added a check for non-empty zones if zoneTopology is non-empty. I don't want to enforce the check explicitly against DCGroup as the number of options for this field may change in the future and this way the minimum number of changes is just to the defaulting logic